src/js/providers/utils/play-promise.ts

```diff
@@ -1,41 +1,34 @@
-export default function createPlayPromise(video) {
-    return new Promise(function(resolve, reject) {
+export default function createPlayPromise(video: HTMLVideoElement): Promise<void> {
+    return new Promise(function(resolve: () => void, reject: (reason: DOMException) => void): void {
         if (video.paused) {
-            return reject(playPromiseError('NotAllowedError', 0, 'play() failed.'));
+            return reject(new DOMException('play() failed.', 'NotAllowedError'));
         }
-        const removeEventListeners = function() {
+        const removeEventListeners = function(): void {
             video.removeEventListener('play', playListener);
             video.removeEventListener('playing', listener);
             video.removeEventListener('pause', listener);
             video.removeEventListener('abort', listener);
             video.removeEventListener('error', listener);
         };
-        const playListener = function() {
+        const playListener = function(): void {
             video.addEventListener('playing', listener);
             video.addEventListener('abort', listener);
             video.addEventListener('error', listener);
             video.addEventListener('pause', listener);
         };
-        const listener = function(e) {
+        const listener = function(e: Event): void {
             removeEventListeners();
             if (e.type === 'playing') {
                 resolve();
             } else {
                 const message = `The play() request was interrupted by a "${e.type}" event.`;
                 if (e.type === 'error') {
-                    reject(playPromiseError('NotSupportedError', 9, message));
+                    reject(new DOMException(message, 'NotSupportedError'));
                 } else {
-                    reject(playPromiseError('AbortError', 20, message));
+                    reject(new DOMException(message, 'AbortError'));
```

Comment:
No more error code? Browsers that need this polyfill do not know what those Error types are or what codes to assign